server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java

```diff
@@ -981,6 +990,123 @@ public void testRestoreShrinkIndex() throws Exception {
+            .setType("fs")
+            .setSettings(Settings.builder().put("location", repoPath)));
+
+        logger.info("--> verify the repository");
+        VerifyRepositoryResponse verifyResponse = client.admin().cluster().prepareVerifyRepository(repositoryName).get();
+        assertThat(verifyResponse.getNodes().length, equalTo(cluster().numDataAndMasterNodes()));
+
+        logger.info("--> create a snapshot");
+        CreateSnapshotResponse snapshotResponse = client.admin().cluster().prepareCreateSnapshot(repositoryName, snapshot0)
+            .setIncludeGlobalState(true)
+            .setWaitForCompletion(true)
+            .execute().actionGet();
+        assertThat(snapshotResponse.status(), equalTo(RestStatus.OK));
+        ensureSearchable();
+
+        SnapshotsStatusResponse response = client.admin().cluster().prepareSnapshotStatus(repositoryName)
+            .setSnapshots(snapshot0)
+            .execute()
+            .actionGet();
+
+        List<SnapshotStatus> snapshots = response.getSnapshots();
+
+        List<Path> files = scanSnapshotFolder(repoPath);
+        assertThat(snapshots, hasSize(1));
+        SnapshotStats stats = snapshots.get(0).getStats();
+
+
+        assertThat(stats.getTotalFileCount(), is(files.size()));
+        assertThat(stats.getTotalSize(), is(files.stream().mapToLong(f -> {
+            try {
+                return Files.size(f);
+            } catch (IOException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }).sum()));
+
+        assertThat(stats.getIncrementalFileCount(), equalTo(stats.getProcessedFileCount()));
+        assertThat(stats.getIncrementalSize(), equalTo(stats.getProcessedSize()));
+
+        // add few docs - less than initially
+        docs = between(1, 5);
+        for (int i = 0; i < docs; i++) {
+            client.prepareIndex(indexName, "type").setSource("test", "test" + i).execute().actionGet();
+        }
+
+        // create another snapshot and drop 1st one
+        // total size has to grow, and has to be equal to files on fs
+        assertThat(client.admin().cluster()
+                .prepareCreateSnapshot(repositoryName, snapshot1)
+                .setWaitForCompletion(true).get().status(),
+            equalTo(RestStatus.OK));
+
+        assertTrue(client.admin().cluster()
+            .prepareDeleteSnapshot(repositoryName, snapshot0)
+            .get().isAcknowledged());
+
+        response = client.admin().cluster().prepareSnapshotStatus(repositoryName)
+            .setSnapshots(snapshot1)
+            .execute()
+            .actionGet();
+
+        final List<Path> files1 = scanSnapshotFolder(repoPath);
+
+        snapshots = response.getSnapshots();
+
+        SnapshotStats anotherStats = snapshots.get(0).getStats();
+
+        assertThat(anotherStats.getIncrementalFileCount(), equalTo(anotherStats.getProcessedFileCount()));
```

Comment:
can you also verify that the incremental file count / size makes sense? This could be established by looking at which files have been added on the filesystem after creating the second snapshot.

Reply:
This comment was not addressed, can you add more assertions here?